### PR TITLE
Refactor <fields> from RzList to RzPVector in RzBinPlugin

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

CI is green.

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/3775 and https://github.com/rizinorg/rizin/issues/3755

### CODE REVIEW

File: librz/bin/p/bin_coff.c [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <rz_util/ht_uu.h>
 
 #include "coff/coff.h"
+#include "rz_vector.h"
```
Comment (wargio) [inline librz/bin/p/bin_coff.c]:
```suggestion
#include <rz_vector.h>
```

**Review by wargio — CHANGES_REQUESTED** [review]
.

File: librz/bin/p/bin_coff.c (hunk 1 above)
Comment (PeiweiHu) [reply librz/bin/p/bin_coff.c]: Maybe this header file is not needed to declare here. It was added by my neovim accidentally. I will remove it later.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@PeiweiHu 2 broken tests, please also rebase

**Review by wargio — APPROVED** [review]

**Pull request merged**